components/eam/src/physics/crm/rrtmgp/radiation.F90

```diff
@@ -1438,8 +1363,8 @@ subroutine radiation_tend(state_in,ptend,    pbuf,          cam_out, cam_in,  &
                ! do the shortwave radiative transfer during the daytime to save
                ! computational cost (and because RRTMGP will fail for cosine solar zenith
                ! angles less than or equal to zero)
+               day_indices_packed = 0
                call set_daynight_indices(coszrs(1:ncol), day_indices(1:ncol), night_indices(1:ncol))
-               nday = count(day_indices(1:ncol) > 0)
```

Comment:
Can you help me understand why you deleted "nnight" in the shortwave driver but deleted "nday" here?

Reply:
Yeah good question. I removed `nday` because it is not used in this context, and I wanted to use `nday` to represent the number of day columns in the packed version of the arrays. `nnight` is still computed here because it is used below in the aerosol optics codes...but admittedly I should have maybe moved it's initialization closer to its use below. `nnight` is not used in the sw section, so no need to include it there. It only lives for the call to the aerosol sw optics driver.

Reply:
ok, it's not important, so no need to change it. It's still clear what the code is doing when you read through it.